### PR TITLE
feat(libflux/semantic): update semantic Flatbuffers schema to match rust semantic graph more closely

### PR DESCRIPTION
**Pull request by fchikwekwe**
### Done checklist
- [ ] docs/SPEC.md updated
- [ ] Test cases written

### CODE REVIEW

File: semantic/semantic.fbs [hunk 1]
```diff
@@ -155,24 +155,14 @@ table ArrayExpression {
 table FunctionExpression {
     loc:SourceLocation;
     defaults:ObjectExpression;
```
Comment (wolffcm) [inline semantic/semantic.fbs]: I think in Rust the function expression does not have a `defaults` field, they are attached to the parameter instead.

Comment (wolffcm) [reply semantic/semantic.fbs]: Also, don't you want to put `params:[FunctionParameter]` in here?

File: semantic/semantic.fbs [hunk 2]
```diff
@@ -155,24 +155,14 @@ table ArrayExpression {
...
-    loc:SourceLocation;
-    parameters:FunctionParameters;
-    body:Block;
-}
-
-table FunctionParameters {
-    loc:SourceLocation;
-    list:[FunctionParameter];
-    pipe:Identifier;
+    Block:Block;
```
Comment (wolffcm) [inline semantic/semantic.fbs]: [nit] the field names in the rest of this file are lower case.

File: semantic/semantic.fbs (hunk 1 above)
Comment (fchikwekwe) [reply semantic/semantic.fbs]: Yeah, I rebased and some of my changes were dropped. I can fix that.

File: semantic/semantic.fbs [hunk 3]
```diff
@@ -267,11 +252,6 @@ table IdentifierExpression   {
     name:string;
 }
 
-table Identifier {
-    loc:SourceLocation;
-    name:string;
-}
-
```
Comment (jlapacik) [inline semantic/semantic.fbs]: Why was `Identifier` removed? `Identifier` and `IdentifierExpression` are distinct concepts. For instance, `IdentifierExpression` is an actual expression and therefore has a type associated with it. The `Identifier` associated with an object property or a function parameter however does not.

Comment (wolffcm) [reply semantic/semantic.fbs]: @jlapacik Yes, you are right.  I didn't catch this when I reviewed.  I'll make another PR that fixes this.

**Pull request merged**